exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_data.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "opentelemetry/sdk/common/circular_buffer.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "vector"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A wrapper class holding in memory exporter data
+ */
+class InMemorySpanData final
+{
+public:
+  InMemorySpanData(size_t);
```

Comment:
I think one of the easiest solutions to make this work would be to make this exporter a header-only implementation.

Could you move all the implementation into the header files? Then circular dependencies shouldn't be a problem.

Reply:
Done